homeassistant/helpers/entity_registry.py

```diff
@@ -902,15 +937,46 @@ def async_get_or_create(
         entity_registry_id: str | None = None
         created_at = utcnow()
         deleted_entity = self.deleted_entities.pop((domain, platform, unique_id), None)
+        options: Mapping[str, Mapping[str, Any]] | None
         if deleted_entity is not None:
-            # Restore id
-            entity_registry_id = deleted_entity.id
+            aliases = deleted_entity.aliases
+            area_id = deleted_entity.area_id
+            categories = deleted_entity.categories
             created_at = deleted_entity.created_at
-
-        entity_id = self.async_generate_entity_id(
-            domain,
-            suggested_object_id or f"{platform}_{unique_id}",
-        )
+            device_class = deleted_entity.device_class
+            disabled_by = (
+                deleted_entity.disabled_by
+                if deleted_entity.disabled_by == RegistryEntryDisabler.USER
+                else disabled_by
+            )
```

Comment:
Maybe we should always restore `disabled_by` and `hidden_by` flags, also if the integration has changed?
That would match the behavior for live entities (we never change the disabled or hidden flags for an entity which has already been seen).